controllers/job.go

```diff
@@ -65,16 +65,27 @@ func unpackContainer(id, containerImage, pullImage string, pullOptions buildv1al
 	}
 }
 
-func createImageContainer(containerImage string, pushOptions buildv1alpha1.Push) v1.Container {
+func pushImageName(artifact buildv1alpha1.OSArtifact) string {
+	pushName := artifact.Spec.PushOptions.ImageName
+	if pushName != "" {
+		return pushName
```

Comment:
I'd find this confusing, e.g. we would output the same image name. Maybe we can use the spec Name instead? or that requires validation maybe?

Reply:
my, bad this is looking good

Reply:
We clarified this on chat. Looks good.